pkg/prometheus/promcfg.go

```diff
@@ -2290,6 +2292,13 @@ func (cg *ConfigGenerator) appendScrapeProtocols(slice yaml.MapSlice) yaml.MapSl
 	return cg.WithMinimumVersion("2.49.0").AppendMapItem(slice, "scrape_protocols", cpf.ScrapeProtocols)
 }
 
+func (cg *ConfigGenerator) appendRuntime(slice yaml.MapSlice, runtime *monitoringv1.RuntimeConfig) yaml.MapSlice {
+	if runtime != nil {
+		return cg.WithMinimumVersion("2.53.0").AppendMapItem(slice, "runtime", runtime)
+	}
+	return slice
```

Comment:
(nit) I'd prefer to build the config explicitly and not rely on the CRD serialization.

```suggestion
	if runtime = nil {
		return slice
	}
	
	var runtimeSlice yaml.MapSlice
	if runtime.GoGC != nil {
		runtimeSlice = cg.WithMinimumVersion("2.53.0").AppendMapItem(runtimeSlice, "gogc", *runtime.GoGC)
	}
	
	return cg.WithMinimumVersion("2.53.0").AppendMapItem(slice, "runtime", runtimeSlice)
```